packages/ubuntu_desktop_installer/lib/pages/who_are_you_page.dart

```diff
@@ -0,0 +1,328 @@
+import '../widgets/validated_input.dart';
+import 'wizard_page.dart';
+
+/// The installer page for setting up the user data.
+///
+/// It uses [WizardPage] and [WizardAction] to create an installer page.
+class WhoAreYouPage extends StatefulWidget {
+  /// Creates a the installer page for setting up the user data.
+  const WhoAreYouPage({Key? key}) : super(key: key);
+
+  @override
+  _WhoAreYouPageState createState() => _WhoAreYouPageState();
+}
+
+/// An enum for storing the login strategy.
+enum LoginStrategy {
+  /// If selected can be used for the representation
+  /// of the user preference to always entering the password
+  /// at login.
+  REQUIRE_PASSWORD,
+
+  /// If selected can be used for the representation
+  /// of the user preference to log in without a password.
+  AUTO_LOGIN
+}
+
+class _WhoAreYouPageState extends State<WhoAreYouPage> {
+  late LoginStrategy _loginStrategy;
+  late TextEditingController _realNameController;
+  late TextEditingController _computerNameController;
+  late TextEditingController _usernameController;
+  late TextEditingController _passwordController;
+  late TextEditingController _confirmPasswordController;
+
+  final _usernameFormKey = GlobalKey<FormState>();
+  final _realNameFormKey = GlobalKey<FormState>();
+  final _computerNameFormKey = GlobalKey<FormState>();
+  final _passwordFormKey = GlobalKey<FormState>();
+  final _confirmPasswordFormKey = GlobalKey<FormState>();
+
+  @override
+  void initState() {
+    _realNameController = TextEditingController();
+    _computerNameController = TextEditingController();
+    _usernameController = TextEditingController();
+    _passwordController = TextEditingController();
+    _confirmPasswordController = TextEditingController();
+
+    _loginStrategy = LoginStrategy.REQUIRE_PASSWORD;
+    super.initState();
+  }
+
+  Widget findPassWordStrengthLabel(String value, String weakPasswordLabel,
+      String averagePasswordLabel, String strongPasswordLabel) {
+    final _strongPattenPasswordValidator =
+        PatternValidator(r'(?=.*?[#?!@$%^&*-])', errorText: '');
+    final _goodPatternPasswordValidator =
+        PatternValidator(r'(^.*(?=.{6,})(?=.*\d).*$)', errorText: '');
+    if (_strongPattenPasswordValidator.isValid(value)) {
+      return Text(strongPasswordLabel,
+          style: TextStyle(
+            color: yaru.Colors.green,
+          ));
+    } else if (_goodPatternPasswordValidator.isValid(value)) {
+      return Text(averagePasswordLabel,
+          style: TextStyle(
+            color: yaru.Colors.orange,
+          ));
+    }
+    if (MultiValidator([
+      RequiredValidator(errorText: ''),
+      MinLengthValidator(2, errorText: ''),
+    ]).isValid(value)) {
+      return Text(weakPasswordLabel,
+          style: TextStyle(
+            color: yaru.Colors.red,
+          ));
+    }
+
+    return Text('');
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const _screenFactor = 1.6;
```

Comment:
> DON’T use a leading underscore for identifiers that aren’t private.
> ...
> There is no concept of “private” for local variables, ...

https://dart.dev/guides/language/effective-dart/style#dont-use-a-leading-underscore-for-identifiers-that-arent-private